horreum-backend/src/test/java/io/hyperfoil/tools/horreum/svc/RunServiceTest.java

```diff
@@ -75,6 +83,75 @@ public void testDataSetCreated(TestInfo info) throws InterruptedException {
       }
    }
 
+   @org.junit.jupiter.api.Test
+   public void testFindSchema(TestInfo info) throws JacksonException {
+      String raw = "{\"value\":42.0,\"$schema\":\"urn:io.hyperfoil.tools.horreum.svc.AlertingServiceTest:testNotifications(TestInfo):1.0\"}";
+      ObjectMapper mapper = new ObjectMapper();
+      JsonNode jsonNode = mapper.readTree(raw);
+      assertNotNull(jsonNode);
+      assertEquals(2, jsonNode.size());
+      assertFalse(jsonNode.path("$schema").isMissingNode());
+      String uri = jsonNode.path("$schema").asText();
+      assertEquals("urn:io.hyperfoil.tools.horreum.svc.AlertingServiceTest:testNotifications(TestInfo):1.0", uri);
+   }
+
+   @org.junit.jupiter.api.Test
+   public void testStraightforwardTransformation(TestInfo info) throws InterruptedException {
+      BlockingQueue<DataSet> dataSetQueue = eventConsumerQueue(DataSet.class, DataSet.EVENT_NEW);
+      NamedJsonPath path = createExampleNamedJsonPath("foo", "$.value");
+      Transformer transformer = createExampleTransformerWithJsonPath(path);
+      Test exampleTest = createExampleTest(getTestName(info));
+      exampleTest.transformers.add(transformer);
+      Test test = createTest(exampleTest);
+      Schema schema = createExampleSchema(info);
+      uploadRun(schema, test.name);
+
+      try (CloseMe ignored = roleManager.withRoles(em, Arrays.asList(TESTER_ROLES))) {
```

Comment:
Wrapping with `withRoles` is necessary only if you execute queries directly against the DB.